tests/cpp/test_no_op.cpp

```diff
@@ -200,4 +204,33 @@ TEST_F(NoOpTest, View) {
   EXPECT_EQ(group->heuristic(), ScheduleHeuristic::NoOp);
 }
 
+TEST_F(NoOpTest, ExpandedReduction) {
+  auto fusion = std::make_unique<Fusion>();
+  FusionGuard fg(fusion.get());
+
+  TensorView* in = TensorViewBuilder()
+                       .ndims(2)
+                       .dtype(DataType::Float)
+                       .contiguity({std::nullopt, std::nullopt})
+                       .shape({2, 3})
+                       .expanded({true, true})
+                       .build();
+  fusion->addInput(in);
+  TensorView* out = sum(in, {0});
+  out = segment_set(out);
+  fusion->addOutput(out);
```

Comment:
Shouldn't this fusion simplify to something like this?
```c++
auto out = mul(2, squeeze(in, {0}));
```
In such case shouldn't this be accepted by `Pointwise` instead of `NoOp`? Or is it smart enough to figure out that this is actually a zero-dimensional problem instead of 1-dimensional? Does `NVFUSER_DUMP=cuda_kernel` show a non-trivial kernel? If so is this similar to #2241?

Reply:
> Shouldn't this fusion simplify to something like this?

Yes

> In such case shouldn't this be accepted by Pointwise instead of NoOp? 

Maybe -- I was surprised to see it no-op too. The test case managed to pass the test [here](https://github.com/NVIDIA/Fuser/blob/446e24737cae95154aca5126a038a30b636ad512/csrc/scheduler/no_op.cpp#L83), so it seems to work as intended. The CUDA kernel generated is the following and yes it looks similar to #2241. 

```
[ RUN      ] NoOpTest.ExpandedReduction

======= Codegen output for kernel: nvfuser_no_op_f0_c1_r0_g0 =======

__global__ void nvfuser_no_op_f0_c1_r0_g0(Tensor<float, 2, 2> T0, Tensor<float, 1, 1> T3) {
  float T1[1];
  T1[0] = 0;
  T1[0]
     = T0[0];
  float T2[1];
  T2[0]
    = T1[0]
    * 2.000000000e+00f;
  T3[0]
     = T2[0];
}

======================================

[       OK ] NoOpTest.ExpandedReduction (164 ms)
```